mne_bids/read.py

```diff
@@ -74,6 +75,48 @@ def _read_raw(raw_fpath, electrode=None, hsp=None, hpi=None, config=None,
     return raw
 
 
+def _handle_info_reading(sidecar_fname, raw):
+    """Read associated sidecar.json and populate raw.
+
+    Handle PowerLineFrequency of recording.
+    """
+    with open(sidecar_fname, "r") as fin:
+        sidecar_json = json.load(fin)
+
+    line_freq = sidecar_json.get("PowerLineFrequency")
+    if line_freq is None and raw.info["line_freq"] is None:
+        # defaults to 50 Hz if neither raw, or sidecar json
+        # has Power Line Freq set
+        raw.info["line_freq"] = 50
+        warn("Neither sidecar json, "
+             "or Raw has Power Line "
+             "Frequency. Defaulting "
+             "to 50 Hz.")
+    elif raw.info["line_freq"] is None and line_freq is not None:
+        # if the read in frequency is not set inside Raw
+        # -> set it to what the sidecar JSON specifies
+        raw.info["line_freq"] = line_freq
+    elif raw.info["line_freq"] is not None \
+            and line_freq is not None \
+            and line_freq != 'n/a':
+        # if both have a set Power Line Frequency, then
+        # check that they are the same, else there is a
+        # discrepency in the metadata of the dataset.
+        if raw.info["line_freq"] != line_freq:
+            raise ValueError("Line frequency in sidecar json does "
+                             "not match the info datastructure of "
+                             "the mne.Raw. "
+                             "Raw is -> {} ".format(raw.info["line_freq"]),
+                             "Sidecar JSON is -> {} ".format(line_freq))
+    else:
+        warn("Line frequency in sidecar json "
+             "is not set. "
+             "Defaulting to what the read in "
+             "data snapshot has.")
```

Comment:
I wouldn't bother with this warning. I think it's safe to assume that if either the BIDS or the raw file has the info but not the other, then it's correct.